awscli overriden dependency pyyaml tests are failing

## Issue description

The awscli derivation was failing on master, because the tests of `pythonPackages.pyyaml` were failing. @FRidh do we attempt to fix the tests of an overridden dependency?

https://github.com/NixOS/nixpkgs/blob/8965de1199f6dbb428be8b31229d1602c6c461ca/pkgs/tools/admin/awscli/default.nix#L24-L32

<details>
<summary>console log</summary>
```
installing
/build/PyYAML-3.13/dist /build/PyYAML-3.13
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
Processing ./PyYAML-3.13-cp27-cp27mu-linux_x86_64.whl
Installing collected packages: PyYAML
Successfully installed PyYAML-3.13
/build/PyYAML-3.13
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/622n69g1sijc2fp7415vl0wyn25hmp7g-python2.7-PyYAML-3.13
shrinking /nix/store/622n69g1sijc2fp7415vl0wyn25hmp7g-python2.7-PyYAML-3.13/lib/python2.7/site-packages/_yaml.so
strip is /nix/store/mgdjnsrkqgmxqjaqaxgqyqm7fwyi96fk-binutils-2.31.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/622n69g1sijc2fp7415vl0wyn25hmp7g-python2.7-PyYAML-3.13/lib
patching script interpreter paths in /nix/store/622n69g1sijc2fp7415vl0wyn25hmp7g-python2.7-PyYAML-3.13
checking for references to /build/ in /nix/store/622n69g1sijc2fp7415vl0wyn25hmp7g-python2.7-PyYAML-3.13...
running install tests
running test
running build_py
running build_ext
skipping 'ext/_yaml.c' Cython extension (up-to-date)
......................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................F..................................................................................F...................................................................................................................................................................................................................................................................................................................................................................................................................................................................
===========================================================================
test_loader_error_ext(tests/data/colon-in-flow-context.loader-error): FAILURE
Traceback (most recent call last):
  File "tests/lib/test_appliance.py", line 65, in execute
    function(verbose=verbose, *filenames)
  File "tests/lib/test_yaml_ext.py", line 240, in wrapper
    function(*args, **kwds)
  File "tests/lib/test_errors.py", line 11, in test_loader_error
    raise AssertionError("expected an exception")
AssertionError: see below
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
'expected an exception'
---------------------------------------------------------------------------
tests/data/colon-in-flow-context.loader-error:
{ foo:bar }
===========================================================================
test_loader_error_string_ext(tests/data/colon-in-flow-context.loader-error): FAILURE
Traceback (most recent call last):
  File "tests/lib/test_appliance.py", line 65, in execute
    function(verbose=verbose, *filenames)
  File "tests/lib/test_yaml_ext.py", line 240, in wrapper
    function(*args, **kwds)
  File "tests/lib/test_errors.py", line 22, in test_loader_error_string
    raise AssertionError("expected an exception")
AssertionError: see below
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
'expected an exception'
---------------------------------------------------------------------------
tests/data/colon-in-flow-context.loader-error:
{ foo:bar }
===========================================================================
TESTS: 2573
FAILURES: 2
error: Tests failed
builder for '/nix/store/lw2j2hldm27fw2yaz0rqi56slq7qaca8-python2.7-PyYAML-3.13.drv' failed with exit code 1
error: build of '/nix/store/lw2j2hldm27fw2yaz0rqi56slq7qaca8-python2.7-PyYAML-3.13.drv' on 'ssh://builder@13.56.251.157' failed: builder for '/nix/store/lw2j2hldm27fw2yaz0rqi56slq7qaca8-python2.7-PyYAML-3.13.drv' failed with exit code 1
builder for '/nix/store/lw2j2hldm27fw2yaz0rqi56slq7qaca8-python2.7-PyYAML-3.13.drv' failed with exit code 1
cannot build derivation '/nix/store/k68g6m7vngm4insr446rlzp336gglz39-awscli-1.16.170.drv': 1 dependencies couldn't be built
error: build of '/nix/store/k68g6m7vngm4insr446rlzp336gglz39-awscli-1.16.170.drv' failed
```
</details>

### Steps to reproduce

Revert https://github.com/NixOS/nixpkgs/pull/65119 and `nix-build -A awscli`

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

 - system: `"x86_64-linux"`
 - host os: `Linux 5.1.11, NixOS, 19.09pre-git (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(root): `""`
 - channels(kalbasit): `"home-manager-18.09, nixpkgs-18.09.932.09195057114"`
 - channels(yl): `""`
 - nixpkgs: `/run/current-system/nixpkgs`

--- a/pkgs/tools/admin/awscli/default.nix
+++ b/pkgs/tools/admin/awscli/default.nix
@@ -2,6 +2,7 @@
 , python
 , groff
 , less
+, fetchpatch
 }:
 
 let
@@ -27,7 +28,11 @@ let
           inherit version;
           sha256 = "3ef3092145e9b70e3ddd2c7ad59bdd0252a94dfe3949721633e41344de00a6bf";
         };
-        doCheck = false;
+        # https://github.com/yaml/pyyaml/issues/298#issuecomment-511990948
+        patches = lib.singleton (fetchpatch {
+          url = "https://github.com/yaml/pyyaml/commit/c5b135fe39d41cffbdc006f28ccb2032df6005e0.patch";
+          sha256 = "0x1v45rkmj194c41d1nqi3ihj9z4rsy8zvpfcd8p960g1fia7fhn";
+        });
       });
     };
   };

